flask-app/.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,51 @@
+name: CI/CD Pipeline for Docker Deployment
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    env:
+      APP_NAME: my-app  # Custom app name (adjust this as needed)
+      DOCKER_IMAGE: ${{ secrets.DOCKER_USERNAME }}/my-app:latest  # DockerHub repo
+
+    steps:
+      # Step 1: Checkout repo
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Step 2: Set up Docker environment
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v2
+
+      # Step 3: Log in to DockerHub
+      - name: Log in to DockerHub
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ secrets.DOCKER_USERNAME }}" --password-stdin
+
+      # Step 4: Build docker image
+      - name: Build and tag Docker image
+        run: |
+          docker build -t ${{ env.DOCKER_IMAGE }} .
+
+      # Step 5: Push Docker image to DockerHub
+      - name: Push Docker image to DockerHub
+        run: |
+          docker push ${{ env.DOCKER_IMAGE }}
+
+      # Step 6: Add SSH key for deployment to the server
+      - name: Add SSH key
+        uses: webfactory/ssh-agent@v0.5.3
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      # Step 7: Deploy Docker image from DockerHub to the server
+      - name: Deploy to server
+        run: |
+          ssh -o StrictHostKeyChecking=no ${{ secrets.SSH_USERNAME }}@${{ secrets.SERVER_IP }} << 'EOF'
```

Comment:
We're not assuming that docker-compose exists on the server, so we have to copy it to the server.